sdk/typescript/src/builder/legacy.ts

```diff
@@ -132,7 +132,14 @@ export async function convertToTransactionBuilder(
           ...(typeof data === 'string' ? fromB64(data) : Array.from(data)),
         ],
       );
-      tx.add(Commands.Publish(modules));
+      const cap = tx.add(Commands.Publish(modules));
```

Comment:
Heads up @666lcz, I ported the change I made to the programmable transaction builder in Rust to the migration logic from legacy transactions -- a legacy transaction now drops the `UpgradeCap` that now gets produced.